src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -101,9 +101,10 @@ internal class CSharpFormatter
             }
 
             // Sorting ensures we count the marker offsets correctly.
-            projectedDocumentLocations = projectedDocumentLocations.OrderBy(l => l).ToList();
+            // We also want to ensure there are no duplicates to avoid duplicate markers.
+            var filteredLocations = projectedDocumentLocations.Distinct().OrderBy(l => l);
```

Comment:
Sometimes, the start/end of a source mapping could be the same as a line start. Hence the need for distinct.

Reply:
You may actually want to ToList this because IEnumerable is lazy and will result in the re-calculation of all this multiple times in the underlying GetCSharpIndentationCoreAsync each time you iterate over the list